rs/state_manager/tests/state_manager.rs

```diff
@@ -6651,9 +6645,7 @@ fn restore_binary_from_snapshot() {
 
 #[test]
 fn restore_chunk_store_from_snapshot() {
-    let env = StateMachineBuilder::new()
-        .with_canister_snapshots(true)
-        .build();
+    let env = StateMachineBuilder::new().build();
```

Comment:
I guess it's a bit of a style question if we should set this flag explicitly for tests that test its functionality or if we should rely on the default. 

I think removing the explicit setting of the flag like you do here is ok, but there are a few more places that you should probably adjust then, such as [here](https://sourcegraph.com/github.com/dfinity/ic@0951bb4fcb0e1acdcbec6acf6e55702ef72d5abf/-/blob/rs/execution_environment/tests/dts.rs?L190:10-190:33)

Reply:
In the execution team we follow the below strategy:
1. While the feature is under development, we enable it explicitly in tests. This is to avoid unnecessary dependencies on existing tests while developing.
2. When the feature is deployed to production, we switch the default. The idea is that we want new tests that are written to exercise any code paths (if relevant, of course it won't be the case for everything). The feature is in the path of becoming stable so the test coverage should include it as much as possible.
3. Once the feature is considered "stable", the flag is dropped (this is a lot easier if the default in tests has been "on" for a while, or else you might end up with a big change to fix any tests that are broken if you remove the flag in one go).

It works well for us but of course each team can choose something different.